openpaas-james/jmap/extensions/src/main/scala/com/linagora/openpaas/james/jmap/json/KeystoreSerializer.scala

```diff
@@ -0,0 +1,58 @@
+package com.linagora.openpaas.james.jmap.json
+
+import com.linagora.openpaas.encrypted.KeyId
+import com.linagora.openpaas.james.jmap.model.KeystoreSetRequest.KeystoreCreationId
+import com.linagora.openpaas.james.jmap.model.{KeystoreCreationRequest, KeystoreCreationResponse, KeystoreSetRequest, KeystoreSetResponse}
+import eu.timepit.refined.api.{RefType, Validate}
+import eu.timepit.refined.collection.NonEmpty
+import eu.timepit.refined.refineV
+import org.apache.james.jmap.core.SetError.SetErrorDescription
+import org.apache.james.jmap.core.{AccountId, Properties, SetError}
+import org.apache.james.jmap.json.mapWrites
+import play.api.libs.json.{Format, JsError, JsObject, JsResult, JsSuccess, JsValue, Json, Reads, Writes}
+
+class KeystoreSerializer {
+  implicit val accountId: Format[AccountId] = Json.valueFormat[AccountId]
+  implicit val propertiesFormat: Format[Properties] = Json.valueFormat[Properties]
+  implicit val setErrorDescriptionWrites: Writes[SetErrorDescription] = Json.valueWrites[SetErrorDescription]
+  implicit val setErrorWrites: Writes[SetError] = Json.writes[SetError]
+
+  implicit def writeRefined[T, P, F[_, _]](
+                                            implicit writesT: Writes[T],
+                                            reftype: RefType[F]
+                                          ): Writes[F[T, P]] = Writes(value => writesT.writes(reftype.unwrap(value)))
+  implicit def readRefined[T, P, F[_, _]](
+                                           implicit readsT: Reads[T],
+                                           reftype: RefType[F],
+                                           validate: Validate[T, P]
+                                         ): Reads[F[T, P]] =
+    Reads(jsValue =>
+      readsT.reads(jsValue).flatMap { valueT =>
+        reftype.refine[P](valueT) match {
+          case Right(valueP) => JsSuccess(valueP)
+          case Left(error)   => JsError(error)
+        }
+      })
```

Comment:
We should have that (refined reads/writes and common utils shared with filter serializer defined in a package.scala class)

Reply:
Was thinking about it, but it's like in MOB, make it work first, refactor after